uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-calendar.js

```diff
@@ -9,21 +11,352 @@ define([
     templateUrl: ['settings', function (settings) {
       return settings.pathTpl + 'components/my-leave-calendar.html';
     }],
-    controllerAs: 'calendar',
-    controller: ['$log', controller]
+    controllerAs: 'ctrl',
+    controller: ['$log', '$q', 'OptionGroup', 'AbsencePeriod', 'AbsenceType',
+      'Calendar', 'PublicHoliday', 'LeaveRequest', controller]
   });
 
 
-  function controller($log) {
+  function controller($log, $q, OptionGroup, AbsencePeriod, AbsenceType, Calendar, PublicHoliday, LeaveRequest) {
     $log.debug('Component: my-leave-calendar');
 
-    init.call(this);
+    var vm = Object.create(this),
+      dayTypes = [],
+      leaveRequests = [],
+      publicHolidays = [],
+      leaveRequestStatuses = [];
+
+    vm.absencePeriods = [];
+    vm.absenceTypes = [];
+    vm.calendar = {};
+    vm.loading = false;
+    vm.months = ['January', 'February', 'March', 'April', 'May', 'June',
+      'July', 'August', 'September', 'October', 'November', 'December'];
+    //Select current month as default
+    vm.selectedMonths = [vm.months[new Date().getMonth()]];
```

Comment:
I think it would be better to do the selection in the `init` block, just to have all the initialization code in one place